api/src/test/java/uk/gov/dhsc/htbhf/claimant/scheduler/HandleCardPendingCancellationJobTest.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.dhsc.htbhf.claimant.entity.Claim;
+import uk.gov.dhsc.htbhf.claimant.message.MessageQueueClient;
+import uk.gov.dhsc.htbhf.claimant.message.MessageType;
+import uk.gov.dhsc.htbhf.claimant.message.payload.EmailMessagePayload;
+import uk.gov.dhsc.htbhf.claimant.repository.ClaimRepository;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static uk.gov.dhsc.htbhf.claimant.entity.CardStatus.PENDING_CANCELLATION;
+import static uk.gov.dhsc.htbhf.claimant.entity.CardStatus.SCHEDULED_FOR_CANCELLATION;
+import static uk.gov.dhsc.htbhf.claimant.message.EmailTemplateKey.FIRST_NAME;
+import static uk.gov.dhsc.htbhf.claimant.message.EmailTemplateKey.LAST_NAME;
+import static uk.gov.dhsc.htbhf.claimant.message.payload.EmailType.CARD_IS_ABOUT_TO_BE_CANCELLED;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimTestDataFactory.aClaimWithCardStatusAndCardStatusTimestamp;
+
+@ExtendWith(MockitoExtension.class)
+class HandleCardPendingCancellationJobTest {
+
+    @Mock
+    private ClaimRepository claimRepository;
+    @Mock
+    private MessageQueueClient messageQueueClient;
+
+    @InjectMocks
+    private HandleCardPendingCancellationJob job;
+
+    @Test
+    public void shouldSendCardToBeCancelledEmailAndSetCardStatus() {
+        Claim claim = aClaimWithCardStatusAndCardStatusTimestamp(PENDING_CANCELLATION, LocalDateTime.now().minusWeeks(16));
+
+        job.handleCardPendingCancellation(claim);
+
+        assertEmailSent(claim);
+        assertClaimCardStatusUpdated();
+    }
+
+    private void assertEmailSent(Claim claim) {
```

Comment:
Is there a nice way we can combine this with the assertions in `DetermineEntitlementNotificationHandlerTest` as they're basically the same apart from the EmailType?

Reply:
Refactored common code into EmailPayloadAssertions.